### ISSUE TITLE
RevealsMap used with GrantPrerequisiteChargeDrainPower causes Ctd

### ISSUE THREAD

**Issue by MlemandPurrs** [OpenRA/OpenRA#19276]
This consistently crashes on second time activation. Activate, let it drain for a bit, deactivate and re, you will get a ctd: 
```
	GrantPrerequisiteChargeDrainPower@SpySattelite:
		PauseOnCondition: disabled || lowpower
		RequiresCondition: england.player
		Prerequisite: sattelitescan
		OrderName: SpySatteliteActive
		DischargeModifier: 800
		ActiveText: ACTIVE
		AvailableText: READY
		ChargeInterval: 12000
		Icon: gps
		Description: Spy Sattelite
		LongDesc: Reveals the entire map while active\nat the cost of doubled Power consumption.\nLasts for up to 1.5 minute.
		SupportPowerPaletteOrder: 9
	RevealsMap@SpySattelite:
		RequiresCondition: satscan-on
		RevealGeneratedShroud: false
	GrantConditionOnPrerequisite@SpySattelite:
		Prerequisites: sattelitescan
		Condition: satscan-on
	PowerMultiplier@SpySattelite:
		Modifier: 200
		RequiresCondition: satscan-on
	GrantConditionOnFaction@GPSReplacer:
		Condition: england.player
		Factions: england
		ResetOnOwnerChange: True
```
here is .log of crash:
```
Exception of type `System.InvalidOperationException`: Attempting to add duplicate shroud source
   at OpenRA.Traits.Shroud.AddSource(Object key, SourceType type, PPos[] projectedCells) in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Traits\Player\Shroud.cs:line 237
   at OpenRA.Mods.Common.Traits.RevealsMap.TraitEnabled(Actor self) in D:\OpenRAModSDK-20210321\engine\OpenRA.Mods.Common\Traits\RevealsMap.cs:line 67
   at OpenRA.Actor.UpdateConditionState(String condition, Int32 token, Boolean isRevoke) in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Actor.cs:line 564
   at OpenRA.Actor.GrantCondition(String condition) in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Actor.cs:line 581
   at OpenRA.Mods.Common.Traits.GrantConditionOnPrerequisite.PrerequisitesUpdated(Actor self, Boolean available) in D:\OpenRAModSDK-20210321\engine\OpenRA.Mods.Common\Traits\Conditions\GrantConditionOnPrerequisite.cs:line 72
   at OpenRA.Mods.Common.Traits.GrantConditionOnPrerequisiteManager.PrerequisitesAvailable(String key) in D:\OpenRAModSDK-20210321\engine\OpenRA.Mods.Common\Traits\Player\GrantConditionOnPrerequisiteManager.cs:line 76
   at OpenRA.Mods.Common.Traits.TechTree.Watcher.Update(Cache`2 ownedPrerequisites) in D:\OpenRAModSDK-20210321\engine\OpenRA.Mods.Common\Traits\Player\TechTree.cs:line 187
   at OpenRA.Mods.Common.Traits.TechTree.Update() in D:\OpenRAModSDK-20210321\engine\OpenRA.Mods.Common\Traits\Player\TechTree.cs:line 48
   at OpenRA.Mods.Common.Traits.TechTree.ActorChanged(Actor a) in D:\OpenRAModSDK-20210321\engine\OpenRA.Mods.Common\Traits\Player\TechTree.cs:line 43
   at OpenRA.Mods.Cnc.Traits.GrantPrerequisiteChargeDrainPower.DischargeableSupportPowerInstance.Activate(Order order) in D:\OpenRAModSDK-20210321\engine\OpenRA.Mods.Cnc\Traits\SupportPowers\GrantPrerequisiteChargeDrainPower.cs:line 182
   at OpenRA.Mods.Common.Traits.SupportPowerManager.ResolveOrder(Actor self, Order order) in D:\OpenRAModSDK-20210321\engine\OpenRA.Mods.Common\Traits\SupportPowers\SupportPowerManager.cs:line 106
   at OpenRA.Actor.ResolveOrder(Order order) in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Actor.cs:line 436
   at OpenRA.Network.UnitOrders.ResolveOrder(Order order, World world, OrderManager orderManager, Int32 clientId) in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Network\UnitOrders.cs:line 356
   at OpenRA.Network.UnitOrders.ProcessOrder(OrderManager orderManager, World world, Int32 clientId, Order order) in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Network\UnitOrders.cs:line 339
   at OpenRA.Network.OrderManager.Tick() in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Network\OrderManager.cs:line 199
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Game.cs:line 615
   at OpenRA.Game.LogicTick() in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Game.cs:line 647
   at OpenRA.Game.Loop() in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Game.cs:line 814
   at OpenRA.Game.Run() in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Game.cs:line 852
   at OpenRA.Game.InitializeAndRun(String[] args) in D:\OpenRAModSDK-20210321\engine\OpenRA.Game\Game.cs:line 270
   at OpenRA.Launcher.Program.Main(String[] args) in D:\OpenRAModSDK-20210321\engine\OpenRA.Launcher\Program.cs:line 26
```
also, if PauseOnCondition occurs while it was active and in use, the using player's view of map gets stuck in revealed state. When this occurs reactivating it no longer produces a crash (probably because its no longer doing anything anymore)

**Comment by pchote** [comment]
I'm fairly sure this is a duplicate of #19147 which will be fixed by #19148.
